R/smtp_send.R

```diff
@@ -169,38 +222,39 @@ smtp_send <- function(email,
       `-inline` = no_options()
     )
 
-  # Create the vector of arguments related
-  # to file attachments
+  # Create the vector of arguments related to file attachments
   attachment_args_vec <- create_attachment_args_vec(email = email)
 
-  # Clean up arguments and options; create the
-  # vector that's needed for `processx::run()`
+  # Clean up arguments and options; create the vector that's
+  # needed for `processx::run()`
   run_args <-
     run_args %>%
     prune_args() %>%
     create_args_opts_vec() %>%
-    append_attachment_args_vec(
-      attachment_args_vec = attachment_args_vec
-    )
+    append_attachment_args_vec(attachment_args_vec = attachment_args_vec)
+
+  if (dry_run) {
```

Comment:
Split `dry_run` from `echo`, so you can decide each independently.

I'd also add a message for `dry_run` where, instead of "The message was sent successfully", you say "Dry run, the message was not sent." or whatever

Reply:
Now implemented!